homeassistant/components/switch/fritzhome.py

```diff
@@ -0,0 +1,66 @@
+"""
+Support for AVM Fritz!Box fritzhome switch devices.
+
+For more details about this component, please refer to the documentation at
+http://home-assistant.io/components/switch.fritzhome/
+"""
+import logging
+from homeassistant.components.fritzhome import DOMAIN
+from homeassistant.components.switch import (SwitchDevice)
+
+DEPENDENCIES = ['fritzhome']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Fritzhome switch platform."""
+    device_list = hass.data[DOMAIN]
+
+    devices = []
+    for device in device_list:
+        if device.has_switch:
+            devices.append(FritzhomeSwitch(hass, device))
+
+    add_devices(devices)
+
+
+class FritzhomeSwitch(SwitchDevice):
+    """The switch class for Fritzhome switches."""
+
+    def __init__(self, hass, device):
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.